spec/app_support/end_reservation_only_spec.rb

```diff
@@ -0,0 +1,73 @@
+require 'spec_helper'
+
+describe EndReservationOnly, :timecop_freeze do
+  let(:now) { DateTime.now.change(hour: 9, min: 31)  }
+
+  let(:action) { described_class.new }
+  let(:order_detail) { reservation.order_detail }
+
+  describe '#perform' do
+    context 'an unpurchased reservation' do
+      let!(:reservation) { create(:setup_reservation, :yesterday, actual_start_at: 1.hour.ago) }
+
+      it 'does not do anything' do
+        expect { action.perform }.not_to change { reservation.reload.actual_end_at }
+        expect(order_detail.state).to eq('new')
+        expect(order_detail.order_status_id).to be_blank
+      end
+    end
+
+    context 'a new reservation only instrument' do
+      let!(:reservation) do
+        start_at = 30.minutes.ago
+        end_at = 1.minute.ago
+
+        create(:purchased_reservation,
+            product: create(:setup_instrument, min_reserve_mins: 1),
```

Comment:
I would make sure that this instrument explicitly says it's reservation-only (I don't know what the default is).